startleft/iriusrisk.py

```diff
@@ -230,3 +230,76 @@ def validate_otm(self):
         logger.debug(f"--- Schema to validate against ---\n{self.schema.json()}\n--- End of schema ---")
         self.schema.validate(self.otm)
         return self.schema
+
+    def check_otm_ids(self):
+        all_valid_ids = []
+        repeated_ids = []
+        wrong_trustzone_ids = []
+        wrong_component_ids = []
+        wrong_component_parent_ids = []
+        wrong_dataflow_ids = []
+        wrong_dataflow_from_ids = []
+        wrong_dataflow_to_ids = []
+
+        for trustzone in self.otm['trustzones']:
+            if trustzone['id'].strip() == '':
+                wrong_trustzone_ids.append(trustzone['id'])
+            elif trustzone['id'] in all_valid_ids:
+                repeated_ids.append(trustzone['id'])
+            elif trustzone['id'] not in all_valid_ids:
+                all_valid_ids.append(trustzone['id'])
+
+        for component in self.otm['components']:
+            if component['id'].strip() == '':
+                wrong_component_ids.append(component['id'])
+            elif component['id'] in all_valid_ids:
+                repeated_ids.append(component['id'])
+            elif component['id'] not in all_valid_ids:
+                all_valid_ids.append(component['id'])
+
+            if component['parent'] not in all_valid_ids:
+                wrong_component_parent_ids.append(component['parent'])
+
+        for dataflow in self.otm['dataflows']:
+            if dataflow['id'].strip() == '':
+                wrong_dataflow_ids.append(dataflow['id'])
+            elif dataflow['id'] in all_valid_ids:
+                repeated_ids.append(dataflow['id'])
+            elif dataflow['id'] not in all_valid_ids:
+                all_valid_ids.append(dataflow['id'])
+
+            if dataflow['from'] not in all_valid_ids:
+                wrong_dataflow_from_ids.append(dataflow['from'])
+
+            if dataflow['to'] not in all_valid_ids:
+                wrong_dataflow_to_ids.append(dataflow['to'])
+
+
+        if len(wrong_trustzone_ids) > 0:
```

Comment:
All "len(array) > 0" can be replaced by "not array" according to Python style guides: https://www.python.org/dev/peps/pep-0008/
It's just a suggestion, change it only if you also prefer that notation

Reply:
exactly. It has been replaced "len(sth)>0:" by "len(sth):"